src/models/dialogModel.ts

```diff
@@ -0,0 +1,43 @@
+import {computed, observable} from 'mobx';
+
+export enum DialogConditionType {
+  Pretrade,
+  Predeposit
+}
+
+export enum DialogActionType {
+  Submit,
+  Checkbox
+}
+
+export class DialogActionModel {
+  id: string;
+  type: DialogActionType;
+  text: string;
+  @observable done: boolean;
+}
+
+// tslint:disable-next-line:max-classes-per-file
+export class DialogModel {
+  id: string;
+  conditionType?: DialogConditionType;
+  header: string;
+  text: string;
+  @observable actions: DialogActionModel[];
+  @observable visible: boolean;
+
+  @computed
+  get isConfirmed() {
+    return this.actions
+      .filter(
+        (action: DialogActionModel) => action.type === DialogActionType.Checkbox
```

Comment:
Maybe it make sense to add a getter that will return Checkbox actions? I think I saw such code already

Reply:
Actually it's not checkbox actions it's Dialog Action -> Checkbox )